### ISSUE TITLE
Bit hangs infinitely while adding remote from Mac to local server

### ISSUE THREAD

**Issue by ipalindromi** [teambit/bit#970]
Hi, I'm running into an odd error trying to get Bit set up on a local server. I've walked through the doc steps, but keep running into the exact same problem. I see a few clues in the logs, but nothing I've found explains them.

```2018-05-08T02:18:41.719Z - debug: SSH: there is no token configured)
2018-05-08T02:18:41.720Z - debug: SSH: reading ssh key file undefined
```
makes me think the ssh key isn't being used, but `bit config get ssh_key_file` returns                                                                                                                                                  
`~/.ssh/id_rsa`, and I can SSH in fine.

```
2018-05-08T02:13:30.669Z - error: got an error from command add <url>: ComponentNotFound. Error serialized: {"stack":"ComponentNotFound\n    at RemoteScopeNotFound.ExtendableBuiltin (/usr/local/Cellar/bit/0.12.12/libexec/lib/node_modules/bit-bin/dist/error/abstract-error.js:49:28)\n    at RemoteScopeNotFound.AbstractError (/usr/local/Cellar/bit/0.12.12/libexec/lib/node_modules/bit-bin/dist/error/abstract-error.js:78:116)\n    at new RemoteScopeNotFound (/usr/local/Cellar/bit/0.12.12/libexec/lib/node_modules/bit-bin/dist/scope/network/exceptions/remote-scope-not-found.js:29:128)\n    at /usr/local/Cellar/bit/0.12.12/libexec/lib/node_modules/bit-bin/dist/scope/network/ssh/ssh.js:326:15\n    at process._tickCallback (internal/process/next_tick.js:178:7)","name":{"stack":"ComponentNotFound\n    at ComponentNotFound.ExtendableBuiltin (/usr/local/Cellar/bit/0.12.12/libexec/lib/node_modules/bit-bin/dist/error/abstract-error.js:49:28)\n    at ComponentNotFound.AbstractError (/usr/local/Cellar/bit/0.12.12/libexec/lib/node_modules/bit-bin/dist/error/abstract-error.js:78:116)\n    at new ComponentNotFound (/usr/local/Cellar/bit/0.12.12/libexec/lib/node_modules/bit-bin/dist/scope/exceptions/component-not-found.js:29:124)\n    at SSH.errorHandler (/usr/local/Cellar/bit/0.12.12/libexec/lib/node_modules/bit-bin/dist/scope/network/ssh/ssh.js:236:18)\n    at Channel.<anonymous> (/usr/local/Cellar/bit/0.12.12/libexec/lib/node_modules/bit-bin/dist/scope/network/ssh/ssh.js:208:54)\n    at Channel.emit (events.js:187:15)\n    at Channel.<anonymous> (/usr/local/Cellar/bit/0.12.12/libexec/lib/node_modules/bit-bin/node_modules/ssh2/lib/Channel.js:135:18)\n    at Object.onceWrapper (events.js:275:13)\n    at Channel.emit (events.js:187:15)\n    at endReadableNT (_stream_readable.js:1091:14)","name":"ComponentNotFound","code":127},"code":129}
2018-05-08T02:13:32.663Z - error: unpackCommand found on error "SyntaxError: Unexpected token z in JSON at position 0", while paring the following string: zsh:1: command not found: bit

2018-05-08T02:13:32.663Z - error: ssh: failed parsing error as JSON, error: zsh:1: command not found: bit
```
Makes it seem like ZSH is failing to run the `bit` command, but it runs fine when I SSH into the server as the bit user.

Thanks!

## Expected Behavior
Running `bit remote add ssh://bit@[server]:/storage/bin/first-scope` adds remote scope

## Actual Behavior
- CLI asks for username, and accepts it
- CLI asks for password, and hangs forever when receiving it

## Steps to Reproduce the Problem

  1. Install Bit on Debian using NPM 
  2. Install Bit on Mac, using brew
  3. Try to run remote command above

## Specifications

  - Bit version: 0.12.12 (Mac) / 0.12.12 (Ubuntu)
  - Node version: 10.00 / 10.00
  - npm / yarn version: 5.6.0 / 5.6.0
  - Platform: Mac Sierra / Ubuntu
  - Bit compiler (include version):
  - Bit tester (include version):

**Comment by GiladShoham** [comment]
hi @ipalindromi 
I think I know what the problem is.... the remote bit scope is prompting for the approval of sending anonymous usage analytics.
please SSH to the server, and configure bit anonymous usage analytics manually (so bit will stop prompting for it)

```
bit config set analytics_reporting false
bit config set error_reporting false
```

I'm trying to figure out the right way to resolve this, but for now - I'll simply add a note to the docs that will handle it.

**Comment by GiladShoham** [comment]
hi @ipalindromi 
Looking at the new logs you uploaded, I think that there are several possibilities for the issue.

from the logs you attached I see that you get an error saying that Bit command was not found, and that the user running the command is running zsh.
is it possible that you have two different terminals on the server, and only one of them (bash) has bit configured in its profile, and not in zsh (or perhaps it is configured for your user profile and not globally?).
another issue that may cause this error is that you have installed Bit as a root, and the user bit does not have permissions to access the binary.

you can try to switch to the user bit on your server, and see if it can run any of Bit's commands (using the `su` command).

if all is configured properly for the bit user, there is another issue you may face (and may be related to this issue as well), which is a permission issue.
when you run `bit init --bare` on the server you can specify permission group using this (taken from bit init -h):
`-s, --shared <groupname>    add group write permissions to a scope properly`
please validate that you have set the write permissions to the scope properly as well.

If the issue still persist, you can find bit's log on the server here - /root/Library/Caches/Bit/logs/debug.log


helpful resource which might help you is another project we created:
https://github.com/teambit/bit-docker
We use it mostly for debugging purpose but you will see there some configuration for bit remote server.
you can also use our image (based on the same project) from https://hub.docker.com/r/bitteam/bit-docker/

Let me know if you need more help setting this up.

**Comment by ipalindromi** [comment]
Just wanted to report back on this. I still haven't figured out exactly what the issue is, but I was able to get `remote add` working with the docker setup, which makes me think that it's something to do with my server's environment. I checked all the points you mentioned, and as far as I can tell everything has access to Bit as I would expect. For now I'm willing to close the ticket though, as it's probably a misconfiguration on the server. Thanks for your help!
